test/mocks/network/mocks.h

```diff
@@ -421,7 +421,7 @@ class MockListener : public Listener {
   MOCK_METHOD(void, setRejectFraction, (UnitFloat));
 };
 
-class MockConnectionHandler : public ConnectionHandler {
+class MockConnectionHandler : public virtual ConnectionHandler {
```

Comment:
Why add `virtual`?

Reply:
A diamond inherence is introduced in the mock object: https://github.com/envoyproxy/envoy/pull/15842#discussion_r614232093